Index: languages/PEoPLang/models/editor.mps
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- languages/PEoPLang/models/editor.mps	(date 1453387543000)
+++ languages/PEoPLang/models/editor.mps	(revision )
@@ -7,48 +7,84 @@
   </languages>
   <imports>
     <import index="v9cq" ref="r:f6dd1409-542b-4f07-b417-52b06dd8c6a4(PEoPLang.structure)" implicit="true" />
+    <import index="wyt6" ref="6354ebe7-c22a-4a0f-ac54-50b52ab9b065/java:java.lang(JDK/)" implicit="true" />
     <import index="tpck" ref="r:00000000-0000-4000-0000-011c89590288(jetbrains.mps.lang.core.structure)" implicit="true" />
   </imports>
   <registry>
     <language id="18bc6592-03a6-4e29-a83a-7ff23bde13ba" name="jetbrains.mps.lang.editor">
       <concept id="1071666914219" name="jetbrains.mps.lang.editor.structure.ConceptEditorDeclaration" flags="ig" index="24kQdi" />
-      <concept id="1140524381322" name="jetbrains.mps.lang.editor.structure.CellModel_ListWithRole" flags="ng" index="2czfm3">
-        <child id="1140524464360" name="cellLayout" index="2czzBx" />
-      </concept>
       <concept id="1237303669825" name="jetbrains.mps.lang.editor.structure.CellLayout_Indent" flags="nn" index="l2Vlx" />
-      <concept id="1237375020029" name="jetbrains.mps.lang.editor.structure.IndentLayoutNewLineChildrenStyleClassItem" flags="ln" index="pj6Ft" />
-      <concept id="1237385578942" name="jetbrains.mps.lang.editor.structure.IndentLayoutOnNewLineStyleClassItem" flags="ln" index="pVoyu" />
       <concept id="1080736578640" name="jetbrains.mps.lang.editor.structure.BaseEditorComponent" flags="ig" index="2wURMF">
         <child id="1080736633877" name="cellModel" index="2wV5jI" />
       </concept>
+      <concept id="1164824717996" name="jetbrains.mps.lang.editor.structure.CellMenuDescriptor" flags="ng" index="OXEIz">
+        <child id="1164824815888" name="cellMenuPart" index="OY2wv" />
+      </concept>
       <concept id="1149850725784" name="jetbrains.mps.lang.editor.structure.CellModel_AttributedNodeCell" flags="ng" index="2SsqMj" />
-      <concept id="1186414536763" name="jetbrains.mps.lang.editor.structure.BooleanStyleSheetItem" flags="ln" index="VOi$J">
-        <property id="1186414551515" name="flag" index="VOm3f" />
+      <concept id="1165253627126" name="jetbrains.mps.lang.editor.structure.CellMenuPart_AbstractGroup" flags="ng" index="1exORT">
+        <child id="1165253890469" name="parameterObjectType" index="1eyP2E" />
       </concept>
+      <concept id="1088013125922" name="jetbrains.mps.lang.editor.structure.CellModel_RefCell" flags="sg" stub="730538219795941030" index="1iCGBv">
+        <child id="1088186146602" name="editorComponent" index="1sWHZn" />
+      </concept>
+      <concept id="1165420413719" name="jetbrains.mps.lang.editor.structure.CellMenuPart_Generic_Group" flags="ng" index="1ou48o">
+        <child id="1165420413721" name="handlerFunction" index="1ou48m" />
+        <child id="1165420413720" name="parametersFunction" index="1ou48n" />
+      </concept>
+      <concept id="1165420626554" name="jetbrains.mps.lang.editor.structure.CellMenuPart_Generic_Group_Handler" flags="in" index="1ouSdP" />
+      <concept id="1088185857835" name="jetbrains.mps.lang.editor.structure.InlineEditorComponent" flags="ig" index="1sVBvm" />
       <concept id="1139848536355" name="jetbrains.mps.lang.editor.structure.CellModel_WithRole" flags="ng" index="1$h60E">
+        <property id="1140017977771" name="readOnly" index="1Intyy" />
         <reference id="1140103550593" name="relationDeclaration" index="1NtTu8" />
       </concept>
-      <concept id="1073389446423" name="jetbrains.mps.lang.editor.structure.CellModel_Collection" flags="sn" index="3EZMnI">
+      <concept id="1073389214265" name="jetbrains.mps.lang.editor.structure.EditorCellModel" flags="ng" index="3EYTF0">
+        <child id="1164826688380" name="menuDescriptor" index="P5bDN" />
+      </concept>
+      <concept id="1073389446423" name="jetbrains.mps.lang.editor.structure.CellModel_Collection" flags="sn" stub="3013115976261988961" index="3EZMnI">
         <child id="1106270802874" name="cellLayout" index="2iSdaV" />
         <child id="1073389446424" name="childCellModel" index="3EZMnx" />
       </concept>
-      <concept id="1073389577006" name="jetbrains.mps.lang.editor.structure.CellModel_Constant" flags="sn" index="3F0ifn">
+      <concept id="1073389577006" name="jetbrains.mps.lang.editor.structure.CellModel_Constant" flags="sn" stub="3610246225209162225" index="3F0ifn">
         <property id="1073389577007" name="text" index="3F0ifm" />
       </concept>
-      <concept id="1073389658414" name="jetbrains.mps.lang.editor.structure.CellModel_Property" flags="sg" index="3F0A7n" />
-      <concept id="1219418625346" name="jetbrains.mps.lang.editor.structure.IStyleContainer" flags="ng" index="3F0Thp">
-        <child id="1219418656006" name="styleItem" index="3F10Kt" />
-      </concept>
-      <concept id="1073390211982" name="jetbrains.mps.lang.editor.structure.CellModel_RefNodeList" flags="sg" index="3F2HdR" />
+      <concept id="1073389658414" name="jetbrains.mps.lang.editor.structure.CellModel_Property" flags="sg" stub="730538219796134133" index="3F0A7n" />
+      <concept id="1163613035599" name="jetbrains.mps.lang.editor.structure.CellMenuPart_AbstractGroup_Query" flags="in" index="3GJtP1" />
       <concept id="1166049232041" name="jetbrains.mps.lang.editor.structure.AbstractComponent" flags="ng" index="1XWOmA">
         <reference id="1166049300910" name="conceptDeclaration" index="1XX52x" />
       </concept>
     </language>
+    <language id="f3061a53-9226-4cc5-a443-f952ceaf5816" name="jetbrains.mps.baseLanguage">
+      <concept id="1145552977093" name="jetbrains.mps.baseLanguage.structure.GenericNewExpression" flags="nn" index="2ShNRf">
+        <child id="1145553007750" name="creator" index="2ShVmc" />
+      </concept>
+      <concept id="1137021947720" name="jetbrains.mps.baseLanguage.structure.ConceptFunction" flags="in" index="2VMwT0">
+        <child id="1137022507850" name="body" index="2VODD2" />
+      </concept>
+      <concept id="1070475926800" name="jetbrains.mps.baseLanguage.structure.StringLiteral" flags="nn" index="Xl_RD">
+        <property id="1070475926801" name="value" index="Xl_RC" />
+      </concept>
+      <concept id="1068580123155" name="jetbrains.mps.baseLanguage.structure.ExpressionStatement" flags="nn" index="3clFbF">
+        <child id="1068580123156" name="expression" index="3clFbG" />
+      </concept>
+      <concept id="1068580123136" name="jetbrains.mps.baseLanguage.structure.StatementList" flags="sn" stub="5293379017992965193" index="3clFbS">
+        <child id="1068581517665" name="statement" index="3cqZAp" />
+      </concept>
+      <concept id="1107535904670" name="jetbrains.mps.baseLanguage.structure.ClassifierType" flags="in" index="3uibUv">
+        <reference id="1107535924139" name="classifier" index="3uigEE" />
+      </concept>
+    </language>
     <language id="ceab5195-25ea-4f22-9b92-103b95ca8c0c" name="jetbrains.mps.lang.core">
       <concept id="1133920641626" name="jetbrains.mps.lang.core.structure.BaseConcept" flags="ng" index="2VYdi">
         <property id="1193676396447" name="virtualPackage" index="3GE5qa" />
       </concept>
     </language>
+    <language id="83888646-71ce-4f1c-9c53-c54016f6ad4f" name="jetbrains.mps.baseLanguage.collections">
+      <concept id="1237721394592" name="jetbrains.mps.baseLanguage.collections.structure.AbstractContainerCreator" flags="nn" index="HWqM0">
+        <child id="1237721435808" name="initValue" index="HW$Y0" />
+        <child id="1237721435807" name="elementType" index="HW$YZ" />
+      </concept>
+      <concept id="1160600644654" name="jetbrains.mps.baseLanguage.collections.structure.ListCreatorWithInit" flags="nn" index="Tc6Ow" />
+    </language>
   </registry>
   <node concept="24kQdi" id="1Gdzz6r7xO4">
     <property role="3GE5qa" value="Base" />
@@ -57,38 +93,46 @@
       <node concept="3F0ifn" id="1Gdzz6r7xOk" role="3EZMnx">
         <property role="3F0ifm" value="VP:" />
       </node>
-      <node concept="2SsqMj" id="1Gdzz6r7xPf" role="3EZMnx" />
-      <node concept="l2Vlx" id="1Gdzz6r7xOd" role="2iSdaV" />
+      <node concept="1iCGBv" id="1Gdzz6r7xPA" role="3EZMnx">
+        <ref role="1NtTu8" to="v9cq:6K8EDSn5e7k" />
+        <node concept="OXEIz" id="2jnRKNLMlnG" role="P5bDN">
+          <node concept="1ou48o" id="2jnRKNLMlnM" role="OY2wv">
+            <node concept="1ouSdP" id="2jnRKNLMlnO" role="1ou48m">
+              <node concept="3clFbS" id="2jnRKNLMlnQ" role="2VODD2" />
-    </node>
+            </node>
+            <node concept="3GJtP1" id="2jnRKNLMlnS" role="1ou48n">
+              <node concept="3clFbS" id="2jnRKNLMlnU" role="2VODD2">
+                <node concept="3clFbF" id="2jnRKNLMmCR" role="3cqZAp">
+                  <node concept="2ShNRf" id="2jnRKNLMn3Q" role="3clFbG">
+                    <node concept="Tc6Ow" id="2jnRKNLMneS" role="2ShVmc">
+                      <node concept="3uibUv" id="2jnRKNLMntW" role="HW$YZ">
+                        <ref role="3uigEE" to="wyt6:~String" resolve="String" />
-  </node>
+                      </node>
-  <node concept="24kQdi" id="7Ufy5VQFo$o">
-    <property role="3GE5qa" value="Base" />
-    <ref role="1XX52x" to="v9cq:6K8EDSn5d7G" resolve="Module" />
-    <node concept="3F0A7n" id="7Ufy5VQFo$u" role="2wV5jI">
-      <ref role="1NtTu8" to="tpck:h0TrG11" resolve="name" />
+                      <node concept="Xl_RD" id="2jnRKNLMnGU" role="HW$Y0">
+                        <property role="Xl_RC" value="Mod1" />
-    </node>
+                      </node>
+                      <node concept="Xl_RD" id="2jnRKNLMo5u" role="HW$Y0">
+                        <property role="Xl_RC" value="Mod2" />
-  </node>
+                      </node>
-  <node concept="24kQdi" id="7Ufy5VQFkRU">
-    <property role="3GE5qa" value="Configuration" />
-    <ref role="1XX52x" to="v9cq:7Ufy5VQFkPw" resolve="ModuleDefinition" />
-    <node concept="3EZMnI" id="7Ufy5VQFkS0" role="2wV5jI">
-      <node concept="3F0ifn" id="7Ufy5VQFkS2" role="3EZMnx">
-        <property role="3F0ifm" value="Module definition" />
-      </node>
+                    </node>
-      <node concept="3F0A7n" id="7Ufy5VQFkS$" role="3EZMnx">
-        <ref role="1NtTu8" to="tpck:h0TrG11" resolve="name" />
-      </node>
+                  </node>
-      <node concept="l2Vlx" id="7Ufy5VQFkS3" role="2iSdaV" />
-      <node concept="3F2HdR" id="7Ufy5VQFkSR" role="3EZMnx">
-        <ref role="1NtTu8" to="v9cq:7Ufy5VQFkRE" />
-        <node concept="l2Vlx" id="7Ufy5VQFkSU" role="2czzBx" />
-        <node concept="pj6Ft" id="7Ufy5VQFkT3" role="3F10Kt">
-          <property role="VOm3f" value="true" />
-        </node>
+                </node>
-        <node concept="pVoyu" id="7Ufy5VQFkTh" role="3F10Kt">
-          <property role="VOm3f" value="true" />
-        </node>
-      </node>
+              </node>
+            </node>
+            <node concept="3uibUv" id="2jnRKNLMls7" role="1eyP2E">
+              <ref role="3uigEE" to="wyt6:~String" resolve="String" />
+            </node>
+          </node>
+        </node>
+        <node concept="1sVBvm" id="1Gdzz6r7xPC" role="1sWHZn">
+          <node concept="3F0A7n" id="1Gdzz6r7xPU" role="2wV5jI">
+            <property role="1Intyy" value="true" />
+            <ref role="1NtTu8" to="tpck:h0TrG11" resolve="name" />
+          </node>
+        </node>
+      </node>
+      <node concept="2SsqMj" id="1Gdzz6r7xPf" role="3EZMnx" />
+      <node concept="l2Vlx" id="1Gdzz6r7xOd" role="2iSdaV" />
     </node>
   </node>
 </model>
